pyanaconda/modules/common/structures/timezone.py

```diff
@@ -82,3 +82,124 @@ def options(self) -> List[Str]:
     @options.setter
     def options(self, value):
         self._options = value
+
+
+class GeolocationResult(DBusData):
+    """Encapsulates the result from geolocation lookup."""
+
+    def __init__(self):
+        self._status = ""
```

Comment:
The status should be always set.

Reply:
I think part of the problem is using the same class to transport data within the module and then out. Perhaps a named tuple would be better for the first part.

However, even after that, any status will be wrong by design in most situations, since you don't know when constructing the object. The constructor would have to have an additional parameter, but that won't work for the D-Bus transport, so it would have to have some default value, and then it could be overlooked all the same because it's optional. So, I think that's a situation where we just can't have it all.

Perhaps dropping the status would be better?

Reply:
I agree, the status is problematic. However, it was supposed to be used instead of `wait_for_refresh_to_finish`. What do you suggest?

Reply:
I guess the new question is, do we want to offer the consumers an option to *not* wait? If not, maybe the result query could include a timeout parameter: `GetGeolocationResult(wait_timeout=GEOLOC_GUI_TIMEOUT)`, then there's always a wait which could turn out to be immediate, and result, which might turn out to be empty. Status is not needed at all.

Reply:
I don't think we want the timeout to be part of the DBus API. It looks like an implementation detail of the UI. We might want to change it in the future. I think we should expose the task with the `GetGeolocationTask()` method. Anything else would be just a copy of some part of the task API anyway.